portfolio/src/main/java/com/google/sps/servlets/LoginServlet.java

```diff
@@ -0,0 +1,71 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.gson.Gson;
+import com.google.sps.data.Login;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet that handles user logins */
+@WebServlet("/login")
+public class LoginServlet extends HttpServlet {
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Set response type to JSON
+    response.setContentType("application/json");
+
+    // Gets reference to a UserService
+    UserService userService = UserServiceFactory.getUserService();
+
+    // Checks and modify behavior based on status of user log in 
+    if (userService.isUserLoggedIn()) {
```

Comment:
nit: It looks like this if-else block is to figure out a parameter for userEmail and an url. So the if-else block can be reduced to only figure those two out. Once the variables are decided, the instantiation of Login and write to response can be done outside. It's easier for people to understand the differences of the two paths. e.g.
```
if () {
   userEmail = ...
   url = ...;
} else {
   userEmail = null;
   url = ...;
}
Login login = new Login(userService.isUserLoggedIn(), userEmail, url);
```

Reply:
+1 to all of this. You could go even further and remove the if {} else {} entirely. Starting with:

boolean loggedIn = userService.isUserLoggedIn
String userEmail = loggedIn ? userService.getCurrentUser().getEmail() : null